napari/layers/utils/layer_utils.py

```diff
@@ -1072,3 +1488,56 @@ def _unique_element(array: Array) -> Optional[Any]:
     if np.any(array[1:] != el):
         return None
     return el
+
+
+def _get_chunk_size(
```

Comment:
bump

Reply:
I would vote against, because right now in order to make it work for napari we do need the check for TensorStore. For TensorStore if chunks are not specified it returns a `tuple` with elements `None` equal to number of dims. For all checks for `if chunk_size` this passes as it is not `None`. The default for all the others is chunk size being equal to the shape of the array. 

A user might expect here for `TensorStore` to return `tuple` of `None` but for our internals it is better to change it to the shape of the array. Hence we change expected behaviour

Reply:
ok